pkg/destroy/gcp/cloudcontroller.go

```diff
@@ -3,18 +3,34 @@ package gcp
 import (
 	"context"
 	"fmt"
+	"regexp"
+	"strings"
 
-	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/compute/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+type resourceFilterFunc func(string) bool
+
+func (o *ClusterUninstaller) defaultFilterFunc(itemName string) bool {
+	return o.isClusterResource(itemName)
+}
```

Comment:
Rather than `defaultFilterFunc` or the existing `isClusterResource` let's go with something more descriptive, something like `hasClusterIDPrefix`